android/src/main/java/nl/xguard/flic2/Flic2.java

```diff
@@ -171,28 +172,32 @@ public void stopScan() {
 
     @Override
     public void onHostResume() {
+        Context context = getReactApplicationContext();
+        boolean isRunning = isServiceRunning(context, Flic2Service.class);
+        if (isRunning) {
+            Intent intent = new Intent(context, Flic2Service.class);
+            getReactApplicationContext().bindService(intent, mFlic2ServiceConnection, BIND_AUTO_CREATE);
+        }
     }
 
     @Override
     public void onHostPause() {
-        // empty
+        try {
+            Context context = getReactApplicationContext();
+            boolean isRunning = isServiceRunning(context, Flic2Service.class);
+            if (mFlic2ServiceConnection.isServiceConnected() && isRunning) {
+                context.unbindService(mFlic2ServiceConnection);
+            }
+        } catch (Exception e) {
+            Log.w(TAG, "onHostPause() , unbindService", e);
+        }
     }
 
     @Override
     public void onHostDestroy() {
-
-      try {
-        if (mFlic2ServiceConnection.isServiceConnected()) {
-          getReactApplicationContext().unbindService(mFlic2ServiceConnection);
-        }
-
         if (mFlic2ServiceDisposable != null) {
-          mFlic2ServiceDisposable.dispose();
+            mFlic2ServiceDisposable.dispose();
```

Comment:
Verdient deze .dispose not een try-catch of is die veilig?

Reply:
Om de mFlic2ServiceDisposable staat al een nullpointer check, deze is hiermee 'veilig'.
Tevens is altijd mogelijk om dispose aan te roepen zodra het object geïnitialiseerd is, zelfde meerdere keren.

Reply:
👍